dependencies/src/main/java/com/google/cloud/tools/opensource/dependencies/DependencyGraphBuilder.java

```diff
@@ -71,9 +77,19 @@ private static DependencyNode resolveCompileTimeDependencies(Artifact rootDepend
     if (cache.containsKey(key)) {
       return cache.get(key);
     }
-    
-    RepositorySystemSession session = RepositoryUtility.newSession(system);
 
+    DefaultRepositorySystemSession session = RepositoryUtility.newSession(system);
+
+    // These combination of DependencySelector comes from `MavenRepositorySystemUtils.newSession`.
+    // StaticLinkageChecker needs to include 'provided' scope.
+    DependencySelector dependencySelector =
+        new AndDependencySelector(
+            new DependencySelector[] {
+                new ScopeDependencySelector(new String[] {"test"}), // "provided" is not here
```

Comment:
This change is to include "provided" scope.

https://github.com/GoogleCloudPlatform/cloud-opensource-java/issues/176#issuecomment-431473418

Reply:
This comment doesn't quite make sense, probably because I don't know what the argument to the constructor is - is it exclusions? (it would help to make that clear in the code or comments)